src/main/java/tc/oc/pgm/match/MatchManagerImpl.java

```diff
@@ -30,7 +30,8 @@
   private final MapLibrary library;
   private final MapLoader loader;
 
-  private final Map<String, Match> matches;
+  private final Map<String, Match> matchesById;
+  private final Map<String, Match> matchesByWorldName;
```

Comment:
Having 2 different maps that keep references to something as relevant as a Match, sounds like asking for trouble & memory leaks down the line. I think there has to be a better way to handle this than using 2 maps.

Reply:
I'm open to any suggestions.

Reply:
@TheMolkaPL How about `Map<String, String> matchIdByWorldName` ?